### PR TITLE
bug: return correct error code and value for OverQuota users

### LINKED ISSUE #836: Return 403 for over quota users.
Return status code of 403 with content of 14 for users that are overquota.

### PR DESCRIPTION
**Pull request by jrconlin**
Closes #836

## Description
Return status 403 (weave error "14") for users that trigger the Quota limits.

## Testing

Start a server with 

```
SYNC_LIMITS__MAX_QUOTA_LIMIT=10 SYNC_ENABLE_QUOTA=1  ...
```
you can then use a script similar to the one in `tools/examples/put.bash` (mind that the value of `SYNC_MASTER_SECRET` matches whatever you're using for the sync server). Since that script attempts to put more than 10 bytes* it should fail with the correct statuscode and body.

* this script may need to be run a second time if the database is empty.

## Issue(s)

Closes #836 .

### CODE REVIEW

**Comment by pjenvey** [comment]
I think adding an additional integration test now would be worthwhile, see https://github.com/mozilla-services/syncstorage-rs/issues/120#issuecomment-701569093

**Comment by jrconlin** [comment]
True. My thought was to keep that a separate issue and get this version on stage for QA testing tomorrow AM.

File: src/error.rs [hunk 1]
```diff
@@ -142,6 +142,7 @@ impl ApiError {
     }
 
     fn weave_error_code(&self) -> WeaveError {
+        dbg!("### self.kind", self.kind());
```
Comment (pjenvey) [inline src/error.rs:145]: nit: probably don't want to keep this

**Review by pjenvey — DISMISSED** [review]

File: src/error.rs (hunk 1 above)
Comment (jrconlin) [reply src/error.rs:145]: crap, yeah. let me kill it quick.

**Review by pjenvey — APPROVED** [review]

**Pull request merged**